### ISSUE TITLE
Invalid(?) JSON response from server

### ISSUE THREAD

**Issue by JeanMertz** [rust-analyzer/rust-analyzer#932]
I'm not quite sure, but I believe I'm getting an invalid JSON response from the server:

```
DEBUG [gen_lsp_server::msg] < {"jsonrpc":"2.0","method":"textDocument/completion","params":{"position":{"character":6,"line":10},"textDocument":{"uri":"file:///mylib/src/main.rs"}},"id":2}
DEBUG [gen_lsp_server::msg] < {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"text":"/// Some comment here!\n/// with\n/// multiple\n/// ```code\n/// lines\n/// ```\nfn main() {\n    let a = 1 * 6;\n\n    writer(a);\n    wri\n}\n\nfn writer(i: usize) {\n    println!(\"{}\", i);\n}\n"}],"textDocument":{"uri":"file:///mylib/src/main.rs","version":4}}}
DEBUG [gen_lsp_server::msg] > {"jsonrpc":"2.0","id":2,"result":null}
DEBUG [gen_lsp_server::msg] > {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"file:///mylib/src/main.rs"}}
```

The first response (line 3) seems fishy. 

If I look at [the specification](https://microsoft.github.io/language-server-protocol/specification#textDocument_completion), the response format is defined as:

> result: `CompletionItem[]` | `CompletionList` | `null`. If a `CompletionItem[]` is provided it is interpreted to be complete. So it is the same as `{ isIncomplete: false, items }`

And in fact, [the language client](https://github.com/autozimu/LanguageClient-neovim) errors out, saying "data did not match any variant of untagged enum CompletionResponse"

**Comment by kjeremy** [comment]
Can you provide a minimal reproducible sample?

**Comment by matklad** [comment]
That’s the collateral damage from our workaround for 
https://github.com/Microsoft/vscode-languageserver-node/issues/457

**Comment by matklad** [comment]
See 
https://github.com/rust-analyzer/rust-analyzer/pull/903

**Comment by matklad** [comment]
@JeanMertz does this actively break your editor, or is this something you just see in the logs?

**Comment by JeanMertz** [comment]
It _somewhat_ actively breaks my editor.

If I open a file (in Vim), and start typing too quickly, it'll return an error, and from that point on autocompletion no longer works. If I start Vim, give it some time, and then start typing, things keep working.

I haven't dug too deep into this, but it definitely caused _some_ issues, I'm not sure how or why though, but it was more than just something I noticed in the logs, I noticed it because I was trying to solve an actual issue.
